net/ftp-proxy/src/etc/inc/plugins.inc.d/ftpproxy.inc

```diff
@@ -0,0 +1,36 @@
+<?php
+/*
+    Copyright (C) 2016 EURO-LOG AG
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    1. Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+
+    2. Redistributions in binary form must reproduce the above copyright
+       notice, this list of conditions and the following disclaimer in the
+       documentation and/or other materials provided with the distribution.
+
+    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+function ftpproxy_firewall($fw)
+{
+    global $config;
+    if (is_array($config['ftpproxy']['ftpproxies'])) {
```

Comment:
My previous comment wasn't correct formatted.

Ok, I change that to 'if (isset($config...) && is_array($config...))'.
When all proxies are deleted the config contains &lt;ftpproxy&gt;&lt;&#47;ftpproxies&gt;&lt;&#47;ftpproxy&gt;.
So there is no need to add the anchors.

Ah, I see. There is an "OPNsense" element in the config . I'll fix that too.

Concerning the additional ftpproxy element my thought was &lt;pluginname&gt;&lt;list of proxies&gt;&lt;single proxy&gt;. It's not needed here.
Sorry it's my first plugin.